sockeye/layers.py

```diff
@@ -23,6 +24,32 @@
 logger = logging.getLogger(__name__)
 
 
+def activation(data: mx.sym.Symbol, act_type: str) -> mx.sym.Symbol:
+    """
+    Apply custom or standard activation.
+
+    Custom activation types include:
+    - Swish-1, also called Sigmoid-Weighted Linear Unit (SiLU): Ramachandran et
+      al. (https://arxiv.org/pdf/1710.05941.pdf), Elfwing et al.
+      (https://arxiv.org/pdf/1702.03118.pdf)
+    - Gaussian Error Linear Unit (GELU): Hendrycks and Gimpel
+      (https://arxiv.org/pdf/1606.08415.pdf)
+
+    :param data: input Symbol of any shape.
+    :param act_type: Type of activation.
+    :return: output Symbol with same shape as input.
+    """
+    # TODO: Contribute these to MXNet?  For now it appears that registered activation types must be implemented in C++.
```

Comment:
Have you observed increased memory or speed decreases with the GELU?
You could think about implementing these as custom operators to avoid creating multiple nodes in the computation graph if these turn out to work better than the Relu.
At least for swish, the backward implementation is easy, but the savings will be negligible there.

Reply:
Compared to ReLU, decoding with Swish-1 takes 100.5% time and GELU takes 103.5% time.  Swish-1 uses the same amount of memory and GELU uses 110% memory.  Swish-1 is basically free while GELU is more expensive, but so far shows the best perplexity.  If they work consistently better than ReLU, we should definitely look at performance optimization.  I think this base implementation is fine for experiments.

Reply:
sounds good, we can revisit this if and when it becomes a bottleneck.